### PR TITLE
More getter routines for fmsDiagFile_type

### PR DESCRIPTION
**Pull request by ganganoaa**
**Description**
* Adds more TODOs.
* Adds a check to a buffer if it is allocated in subroutine _allocate_diag_field_output_buffers()_.
* Adds getter routines _get_last_output_, _get_next_output_, _get_next_next_output_, and _get_no_more_data_ to the derived type _fmsDiagFile_type_.

Fixes # (issue)

**How Has This Been Tested?**
Please describe the tests that you ran to verify your changes. Please also note
any relevant details for your test configuration (e.g. compiler, OS).  Include
enough information so someone can reproduce your tests.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: diag_manager/fms_diag_file_object.F90 [hunk 1]
```diff
@@ -126,6 +126,10 @@ module fms_diag_file_object_mod
  procedure, public :: get_file_duration_units
  procedure, public :: get_file_varlist
  procedure, public :: get_file_global_meta
+ procedure, public :: get_last_outout
```
Comment (uramirez8707) [inline diag_manager/fms_diag_file_object.F90:129]: this should be `get_last_outout`

File: diag_manager/fms_diag_file_object.F90 [hunk 2]
```diff
@@ -541,6 +545,38 @@ pure function get_file_global_meta (this) result(res)
...
+!! \return Copy of next_output time
+function get_next_outout (this) result(res)
+ class(fmsDiagFile_type), intent(in) :: this !< The file object
+  type(time_type) :: res
+  res = this%next_output
+end function get_next_outout
+
+!> \brief Returns a copy of next_next_output time
+!! \return Copy of next_next_output time
+function get_next_next_outout (this) result(res)
```
Comment (uramirez8707) [inline diag_manager/fms_diag_file_object.F90:566]: typo, this should be `get_next_next_output`

**Review by uramirez8707 — CHANGES_REQUESTED** [review]
You have some typos, which is why the CI is failing

File: diag_manager/fms_diag_object.F90 [hunk 3]
```diff
@@ -29,14 +29,16 @@ module fms_diag_object_mod
 use fms_diag_file_object_mod, only: fmsDiagFileContainer_type, fmsDiagFile_type, fms_diag_files_object_init
 use fms_diag_field_object_mod, only: fmsDiagField_type, fms_diag_fields_object_init, get_default_missing_value
 use fms_diag_yaml_mod, only: diag_yaml_object_init, diag_yaml_object_end, find_diag_field, &
-                           & get_diag_files_id, diag_yaml, get_diag_field_ids, DiagYamlFilesVar_type
+                           & get_diag_files_id, diag_yaml, diagYamlFilesVar_type, get_diag_field_ids
```
Comment (thomas-robinson) [inline diag_manager/fms_diag_object.F90:32]: Why is this changed?

File: diag_manager/fms_diag_object.F90 [hunk 4]
```diff
@@ -553,9 +564,14 @@ logical function fms_diag_accept_data (this, diag_field_id, field_data, time, is
 !!TODO: Loop through fields and do averages/math functions
     do i = 1, size(this%FMS_diag_fields(diag_field_id)%buffer_ids)
       buffer_id = this%FMS_diag_fields(diag_field_id)%buffer_ids(i)
+      file_id = this%FMS_diag_fields(diag_field_id)%file_ids(i)
```
Comment (thomas-robinson) [inline diag_manager/fms_diag_object.F90:567]: Everything you add here needs to also go in `fms_diag_send_complete` because in the `main_if: if (buffer_the_data) then` statement when `buffer_the_data` is `.true.`, all of the math reductions will be done in `fms_diag_send_complete`.  I think you should put as much as possible contained in this `else` block into a function/subroutine to avoid duplicate code.

**Review by thomas-robinson — CHANGES_REQUESTED** [review]

File: diag_manager/fms_diag_object.F90 (hunk 4 above)
Comment (thomas-robinson) [reply diag_manager/fms_diag_object.F90:567]: You could even put this into a new module or object to handle just the math functions.

File: diag_manager/fms_diag_object.F90 (hunk 3 above)
Comment (ganganoaa) [reply diag_manager/fms_diag_object.F90]: caused by merging locally; will be reverted.

File: diag_manager/fms_diag_object.F90 (hunk 4 above)
Comment (ganganoaa) [reply diag_manager/fms_diag_object.F90:567]: You are right. I will move the whole block to a new subroutine.

**Comment by uramirez8707** [comment]
This is included in #1295 and it is now out of date. Closing.
